test/functional/apps/visualize/_area_chart.js

```diff
@@ -62,6 +62,31 @@ bdd.describe('visualize app', function describeIndexTests() {
   bdd.describe('area charts', function indexPatternCreation() {
     var vizName1 = 'Visualization AreaChart';
 
+    bdd.it('should save and load with special characters', function pageHeader() {
+      let vizName2 = vizName1 + '/?&=%';
+      return PageObjects.visualize.saveVisualization(vizName2)
+        .then(function (message) {
+          PageObjects.common.debug('Saved viz message = ' + message);
+          PageObjects.common.saveScreenshot('Visualize-area-chart-save-toast');
+          expect(message).to.be('Visualization Editor: Saved Visualization \"' + vizName2 + '\"');
+        })
+        .then(function testVisualizeWaitForToastMessageGone() {
+          return PageObjects.visualize.waitForToastMessageGone();
+        })
+        .then(function loadSavedVisualization() {
+          return PageObjects.visualize.loadSavedVisualization(vizName2);
+        })
+        .then(function () {
+          return PageObjects.visualize.waitForVisualization();
+        })
+        // We have to sleep sometime between loading the saved visTitle
+        // and trying to access the chart below with getXAxisLabels
+        // otherwise it hangs.
+        .then(function sleep() {
+          return PageObjects.common.sleep(2000);
```

Comment:
I don't think there's really any reason to wait for the toast message, load the viz and wait for it if we're not running assertions on it. I think the only reason the test below does so is to set things up for the test that follows it.